src/sss_client/idmap/sss_nss_idmap.h

```diff
@@ -152,6 +152,48 @@ int sss_nss_getidbysid(const char *sid, uint32_t *id,
 int sss_nss_getorigbyname(const char *fq_name, struct sss_nss_kv **kv_list,
                           enum sss_id_type *type);
 
+/**
+ * @brief Find original data by fully qualified user name
+ *
+ * @param[in] fq_name  Fully qualified name of a user
+ * @param[out] kv_list A NULL terminate list of key-value pairs where the key
+ *                     is the attribute name in the cache of SSSD,
+ *                     must be freed by the caller with sss_nss_free_kv()
+ * @param[out] type    Type of the object related to the given name
+ *
+ * @return
+ *  - 0 (EOK): success
+ *  - ENOENT: requested user was not found in the domain extracted from the given name
+ *  - ENETUNREACH: SSSD does not know how to handle the domain extracted from the given name
+ *  - ENOSYS: this call is not supported by the configured provider
+ *  - EINVAL: input cannot be parsed
+ *  - EIO: remote servers cannot be reached
+ *  - EFAULT: any other error
+ */
+int sss_nss_getorigbyusername(const char *fq_name, struct sss_nss_kv **kv_list,
+                              enum sss_id_type *type);
```

Comment:
`type` must be `SSS_ID_TYPE_UID`, no?

Reply:
Hi,

in general yes, but it might also be `SSS_ID_TYPE_BOTH` to indicate an automatically generated user-private group. Additionally I kept the `type` parameter to allow an easy migration since this way the new functions can be used as a drop-in replacement for `sss_nss_getorigbyname()`.

HTH

bye,
Sumit

Reply:
(For the sake of my education)
Attrs of what object are contained in `kv_list` in the case `type == SSS_ID_TYPE_BOTH`?
Is it still user object and type merely indicates this user object has MGP? Or does `kv_list` contains attrs for both user and MPG?

Does the same apply to sid-by-*id requests?

Reply:
Hi,

Yes, it is a user object and the type indicates that a group with the same name and with the GID of the user exists.

HTH

bye,
Sumit